File: Semestre_4_2019_2020/INFO0402_C++/TP/algos/algo1/main.c
#include <stdio.h>

int main() {
    int taille, i=0, max;
    //Initialisation de taille
    printf("Entrez la taille du tableau:\n");
    scanf("%d", &taille);
    //Initialisation des tableaux
    int T1[taille], T2[taille];
    printf("Entrez les valeurs du tableau:\n");
    while(i<taille){
        scanf("%d", &T1[i]);
        i++;
    }
    //Affichage de T1
    i=0;
    while(i<taille){
        printf("%d \n", T1[i]);
        i++;
    }
    //Recherche du max
    max=T1[0];
    for(i=1; i<taille; i++){
        if(max<T1[i]){
            max=T1[i];
        }
    }
    //Normalisation
    printf("%d \n", max);
    for(i=0; i<taille; i++){
        T2[i] = T1[i]/max;
    }
    //Affichage de T2
    for(i=0; i<taille; i++){
        printf("%d \n", T2[i]);
    }
    return 0;
}

File: Semestre_4_2019_2020/INFO0402_C++/TP/algos/algo1/main.java
import java.util.Scanner;

class main{
    public static void main (String [] args) {
        Scanner clavier = new Scanner(System.in);
        int taille, i=0, max;

        //initialisation de taille
        System.out.println("Entrez la taille du tableau:");
        taille=clavier.nextInt();

        //initialisation des tableaux
        int [] T1= new int [taille];
        int [] T2= new int [taille];
        System.out.println("Entrez les valeurs du tableau:");
        while(i<taille){
            T1[i]=clavier.nextInt();
            i++;
        }

        //affichage de T1
        i=0;
        while(i<taille){
            System.out.println("valeur "+i+" = "+T1[i]+" ");
            i++;
        }

        //recherche du max
        max=T1[0];
        for(i=1;i<taille;i++){
            if(max<T1[i]){
                max=T1[i];
            }
        }

        //normalisation
        System.out.println("max="+max+" ");
        for(i=0;i<taille;i++){
            T2[i]=T1[i]/max;
        }

        //affichage de T2
        for(i=0;i<taille;i++){
            System.out.println(T2[i]+" ");
        }
    }
}